pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -351,38 +353,42 @@ class RpmContentSaver(ContentSaver):
                 :class:`~pulpcore.plugin.stages.DeclarativeContent` objects to be saved.
 
         """
-        update_collection_to_save = []
+        update_collections_to_save = []
         update_references_to_save = []
+        update_collection_packages_to_save = []
+
         for declarative_content in batch:
             if declarative_content is None:
                 continue
             if not isinstance(declarative_content.content, UpdateRecord):
                 continue
             update_record = declarative_content.content
-            try:
-                update_collections = update_record._collections
-            except AttributeError:
-                pass  # This UpdateRecord was found in the db or has no collections or references
-            else:
-                for update_collection in update_collections:
-                    update_collection.update_record = update_record
-                    update_collection_to_save.append(update_collection)
-
-                update_references = update_record._references
-                for update_reference in update_references:
-                    update_reference.update_record = update_record
-                    update_references_to_save.append(update_reference)
 
-        update_collection_packages_to_save = []
-        if update_collection_to_save:
-            saved_collections = UpdateCollection.objects.bulk_create(update_collection_to_save)
-            for update_collection in saved_collections:
-                for update_collection_package in update_collection._packages:
+            relations_exist = update_record.collections.count() or update_record.references.count()
+            if relations_exist:
+                # existing content which was retrieved from the db at earlier stages
+                continue
+
+            future_relations = declarative_content.extra_data
+            update_collections = future_relations.get('collections') or {}
```

Comment:
This also looks like a defaultdict case so the default is specified with the definition instead of here. No change request still tho.

Reply:
It also covers the case when extra_date is not set at all. So I think I'll leave it as-is for now. Thanks